code/datums/components/seethrough_mob.dm

```diff
@@ -0,0 +1,132 @@
+///A component that lets you turn your character transparent in order to see and click through yourself.
+/datum/component/seethrough_mob
+	///The atom that enables our dark magic
+	var/atom/movable/render_source_atom
+	///The fake version of ourselves
+	var/image/trickery_image
+	///Which alpha do we animate towards?
+	var/target_alpha
+	///How long our faze in/out takes
+	var/animation_time
+	///Does this object let clicks from players its transparent to pass through it
+	var/clickthrough
+	///Is the seethrough effect currently active
+	var/is_active
+	///The mob's original render_target value
+	var/initial_render_target_value
+	///This component's personal uid
+	var/personal_uid
+
+/datum/component/seethrough_mob/Initialize(target_alpha = 100, animation_time = 0.5 SECONDS, clickthrough = TRUE)
+	. = ..()
+
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.target_alpha = target_alpha
+	src.animation_time = animation_time
+	src.clickthrough = clickthrough
+	src.is_active = FALSE
+	src.render_source_atom = new()
+	var/datum/action/toggle_seethrough/action = new(src)
+	var/static/uid = 0
+	uid++
+	src.personal_uid = uid
+
+	action.Grant(parent)
+
+/datum/component/seethrough_mob/Destroy(force, silent)
+	QDEL_NULL(render_source_atom)
+	return ..()
+
+///Set up everything we need to trick the client and keep it looking normal for everyone else
+/datum/component/seethrough_mob/proc/trick_mob()
+	SIGNAL_HANDLER
+
+	var/mob/fool = parent
+	var/datum/hud/our_hud = fool.hud_used
+	for(var/atom/movable/screen/plane_master/seethrough in our_hud.get_true_plane_masters(SEETHROUGH_PLANE))
```

Comment:
```suggestion
	for(var/atom/movable/screen/plane_master/seethrough as anything in our_hud.get_true_plane_masters(SEETHROUGH_PLANE))
```
`as anything` should be used if we're sure that every item in a list is of the same type.